File: back-end/src/main/java/es/ull/project/application/service/operation/OperationCreateService.java
package es.ull.project.application.service.operation;

import org.springframework.beans.factory.annotation.Autowired;

import es.ull.project.application.repository.OperationRepository;
import es.ull.project.application.usecase.operation.OperationCreateUseCase;
import es.ull.project.domain.valueobject.Operation;
import es.ull.project.domain.valueobject.OperationName;
import es.ull.utils.docker.UllDockerImageName;

public class OperationCreateService implements OperationCreateUseCase {

    @Autowired
    private OperationRepository repository;

    @Override
    public Operation createOperation(OperationName name, UllDockerImageName dockerImage) {
        Operation existingOperation = repository.fetchByName(name);
        if (existingOperation != null) {
            throw new IllegalArgumentException("Operation with name " + name + " already exists.");
        }
        final Operation newOperation = new Operation(name, dockerImage);
        return repository.save(newOperation);
    }
}

File: front-end/src/application/service/operation/operation-create-service.ts
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import type { OperationCreateUseCase } from '../../usecase/operation/operation-create-use-case';
import { Operation } from '../../../domain/value-object/operation';
import { OperationName } from '../../../domain/value-object/operation-name';
import { UllDockerImageName } from '@ull-tfg/ull-tfg-typescript/dist/ull-docker-image-name';
import type { OperationRepository } from '../../repository/operation-repository';
import { NodeIcon } from '@/domain/enumerate/node-icon';

export class OperationCreateService implements OperationCreateUseCase {

    constructor(private readonly operationRepository: OperationRepository) { }

    async createOperation(name: OperationName, dockerImage: UllDockerImageName): Promise<Either<DataError, Operation>> {
        const operation = new Operation(name, dockerImage); 
        return this.operationRepository.saveOperation(operation); 
    }
}